File: Python/Leetcode/Palindrome Number/C/solution_1.c
#include<stdio.h>
#include<stdbool.h>

bool isPalindrome(int x) {
    if (x < 0) return false;  // Negative numbers are not palindromes

    long long temp = 0;
    long long c = 0;
    long long original = x;

    while (x > 0) {
        c = x % 10;
        x /= 10;
        temp = temp * 10 + c;
    }

    return temp == original;
}

File: Python/Leetcode/Palindrome Number/solution_1.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        # Negative numbers are not palindromes
        if x < 0:
            return False
        
        # Convert the integer to a string to compare the reversed version
        str_x = str(x)
        return str_x == str_x[::-1]
